spec/lib/pedicel/base_spec.rb

```diff
@@ -359,100 +359,106 @@ def create_special_certificate(ca_key, ca_certificate, config, oids)
       expect{Pedicel::Base.verify_x509_chain(params)}.to_not raise_error
     end
 
-    it 'errs if certs are interchanged' do
-      params.keys.permutation.reject{|ks| ks == params.keys}.each do |permutated_keys|
-        permutated_params = permutated_keys.zip(params.values).to_h
+    context 'intermediate equals leaf' do
+      it 'errs if intermediate = leaf (because root did not sign leaf)' do
+        params[:intermediate] = params[:leaf]
+        expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, 'invalid chain due to intermediate')
+      end
 
-        expect{Pedicel::Base.verify_x509_chain(permutated_params)}.to raise_error(Pedicel::SignatureError, /invalid chain/)
+      it 'errs even if intermediate = leaf is self-signed' do
+        params[:intermediate] = params[:leaf] = PedicelPay::Backend.generate.ca_certificate
+        expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, 'invalid chain due to intermediate')
       end
     end
 
-    it 'errs if intermediate equals leaf (because root did not sign leaf)' do
-      params[:intermediate] = params[:leaf]
-      expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, /invalid chain due to intermediate/)
-    end
+    context 'leaf equals intermediate' do
+      it 'errs if leaf = intermediate (because intermediate did not sign leaf)' do
+        params[:leaf] = params[:intermediate]
+        expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, 'invalid chain due to leaf')
+      end
 
-    it 'errs if leaf equals intermediate (because the intermediate must sign leaf)' do
-      params[:leaf] = params[:intermediate]
-      expect{Pedicel::Base.verify_x509_chain(params)}.to raise_error(Pedicel::SignatureError, /invalid chain/)
+      it 'errs even if leaf = intermediate is self-signed' do
```

Comment:
I think having the same test twice is superfluous. Even though it makes sense semantically :)

Reply:
That's true, I'd certainly accept to replace one with a comment. Please feel free to push 🙂